sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
I wanted to avoid applying mariadb defaults all the time. So have to use some values to compare.
According linux source code 0 will cause the error:
https://github.com/torvalds/linux/blob/b29794ec95c6856b316c2295904208bf11ffddd9/net/ipv4/tcp.c#L2612